doc/source/cd/background.md

```diff
@@ -0,0 +1,862 @@
+at the feature level when they occur. The detector is inspired by the
+work of [Jitkrittum et al. (2016)](https://arxiv.org/abs/1605.06796)
+but various major adaptations have been made.
+
+As with the usual classifier-based approach, a portion of the available
+data is used to train a classifier that can discriminate reference
+instances from test instances. However, the spot-the-diff detector is
+specified such that when drift is detected, we can inspect the weights
+of the classifier to shine light on exactly which features of the data
+were used to distinguish reference from test samples, and therefore
+caused drift to be detected. The [Interpretable drift detection with the
+spot-the-diff detector on MNIST and Wine-Quality
+datasets](../examples/cd_spot_the_diff_mnist_wine.nblink) example
+demonstrates this capability.
+
+### Model uncertainty
+
+The [model uncertainty](methods/modeluncdrift.ipynb)-based drift
+detector uses the ML model of interest itself to detect drift. It
+doesn’t fit as neatly under the category of *learned detectors*, since
+unlike the others, it does not require an additional detector/model to be
+trained. Nevertheless, it is included in this category since, like the
+other learned detectors, it is important that the reference data set
+given to the detector is disjoint from the training set used to train
+the model of interest.
+
+Model-uncertainty drift detectors aim to directly detect drift that’s
+likely to effect the performance of a model of interest. The approach is
+to test for change in the number of instances falling into regions of
+the input space on which the model is uncertain in its predictions. For
+each instance in the reference set the detector obtains the model’s
+prediction and some associated notion of uncertainty. The same is done
+for the test set and if significant differences in uncertainty are
+detected (via a Kolmogorov-Smirnoff test) then drift is flagged. The
+model’s notion of uncertainty depends on the type of model. For a
+classifier this may be the entropy of the predicted label probabilities.
+For a regressor with dropout layers, [dropout Monte
+Carlo](http://proceedings.mlr.press/v48/gal16.pdf) can be used to
+provide a notion of uncertainty.
+
+## 4. Online drift detection
+
+So far, we have discussed drift detection in an *offline* context, with
+the entire test set $\{\mathbf{X}_i\}_{i=1}^{N}$ compared to the
+reference dataset $\{\mathbf{X}^{ref}_i\}_{i=1}^{M}$. However, at
+test time data sometimes arrives sequentially, meaning drift detection
+must be performed in an *online* fashion.
+
+```{image} images/online.*
+:align: center
+:alt: Online drift detection
+:width: 350px
+```
+
+One approach is to perform a test for drift every $W$ time-steps,
+using the $W$ samples that have arrived since the last test. In
+other words, that is to compare $\{\mathbf{X}_i\}_{i=t}^{t+W}$ to
+$\{\mathbf{X}^{ref}_i\}_{i=1}^{M}$. Such a strategy could be
```

Comment:
Nitpick but could we use t-W+1 to t rather than t to t+W given those are the observations available at time t? 

If keeping as is the upper limit should be changed to t+W-1